internal/github/checkStalePRs.go

```diff
@@ -0,0 +1,71 @@
+package github
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"strings"
+	"time"
+
+	"github.com/google/go-github/v60/github"
+
+	"github.com/chia-network/github-bot/internal/config"
+)
+
+// CheckStalePRs will return a list of PR URLs that have not been updated in the last 7 days by internal team members.
+func CheckStalePRs(githubClient *github.Client, internalTeam string, cfg config.LabelConfig) ([]string, error) {
+	var stalePRUrls []string
+	cutoffDate := time.Now().AddDate(0, 0, -7) // 7 days ago
+	teamMembers, err := GetTeamMemberList(githubClient, internalTeam)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, fullRepo := range cfg.LabelCheckRepos {
+		log.Println("Checking repository:", fullRepo.Name)
+		parts := strings.Split(fullRepo.Name, "/")
+		if len(parts) != 2 {
+			return nil, fmt.Errorf("invalid repository name - must contain owner and repository: %s", fullRepo.Name)
+		}
+		owner, repo := parts[0], parts[1]
+
+		communityPRs, err := FindCommunityPRs(owner, repo, teamMembers, githubClient)
+		if err != nil {
+			return nil, err
+		}
+
+		for _, pr := range communityPRs {
+			if isStale(githubClient, pr, teamMembers, cutoffDate) {
+				stalePRUrls = append(stalePRUrls, pr.GetHTMLURL()) // Collecting URLs instead of PR objects
+			}
+		}
+	}
+	return stalePRUrls, nil
+}
+
+// Checks if a PR is stale based on the last update from team members
+func isStale(githubClient *github.Client, pr *github.PullRequest, teamMembers map[string]bool, cutoffDate time.Time) bool {
+	// Retrieve the timeline for the PR to find the latest relevant event
+	listOptions := &github.ListOptions{PerPage: 100}
+	for {
+		// Note: As far as the GitHub API is concerned, every pull request is an issue,
+		// but not every issue is a pull request.
+		events, resp, err := githubClient.Issues.ListIssueTimeline(context.TODO(), pr.Base.Repo.Owner.GetLogin(), pr.Base.Repo.GetName(), pr.GetNumber(), listOptions)
+		if err != nil {
+			log.Printf("Failed to get timeline for PR #%d: %v", pr.GetNumber(), err)
+			break // Skip to next PR on error
+		}
+		for _, event := range events {
+			if event.Event != nil && *event.Event == "commented" && teamMembers[*event.Actor.Login] && event.CreatedAt.After(cutoffDate) {
```

Comment:
I'll have to look, but I'm curious if there are other types of events that should count here.

Reply:
Maybe https://docs.github.com/en/rest/using-the-rest-api/issue-event-types?apiVersion=2022-11-28#reviewed

Reply:
Though, I don't know if `reviewed` maybe also implies `commented`? But I don't know if it necessarily does or doesn't

Reply:
Additionally, with regards to reviews, if the last event IS a review, maybe this stale PR should also just be ignored? That might be up for debate tbh. But my logic there is basically, if we're waiting on the PR author to submit requested changes, the ball isn't waiting in our court.

Reply:
Maybe commented or reviewed by a teammember within the last seven days then?

Reply:
Yeah that makes sense to me.

Reply:
Added checks for if a pr has been reviewed or not